peak/peak.py

```diff
@@ -1,2 +1,45 @@
+from collections import OrderedDict
+from bit_vector import BitVector
+
 class Peak:
     pass
+
+
+def name_outputs(**outputs):
+    """Decorator meant to apply to any function to specify output types
+    The output types will be stored in fn.__peak_outputs__
+    The input types will be stored in fn.__peak_inputs__
+    Will verify that all the inputs have type annotations
+    Will also verify that the outputs of running fn will have the correct number of bits
+    """
+    def decorator(call_fn):
+        def call_wrapper(*args,**kwargs):
+            results = call_fn(*args,**kwargs)
+            single_output = not isinstance(results,tuple)
+            if single_output:
+                results = (results,)
+            for i, (oname,otype) in enumerate(outputs.items()):
+                assert results[i].num_bits == otype(0).num_bits, f"{results[i].num_bits} != {otype(0).num_bits}"
```

Comment:
Should thrown a TypeError not raise an assertion.

Test should be `type(result[i]) is otype` after the merge of new BitVector.  Fine for now.